### ISSUE TITLE
Shipping method no longer selected after second payment method change

### ISSUE THREAD

**Issue by shopwareBot** [shopware/platform#4273]
There are two payment methods to choose from in the checkout: Prepayment and self-collection.
The rule builder has been used to specify that the DHL shipping method is selected for the prepayment payment method and the self-pickup (store) shipping method is selected for the self-pickup payment method. Prepayment is stored as the default payment method in the sales channel.

If a customer goes to the checkout, the payment method prepayment and shipping method DHL are selected. 
If the customer now selects the payment method Self pickup, the shipping method changes to Self pickup (store) as expected. 

If the payment method is now changed a second time, back to prepayment, the status message 'The shipping method "Self pickup (store)" is not available for your current shopping cart, the shipping method has been changed to "DHL"' appears, but the DHL shipping method checkbox is no longer selected in the frontend.  Only after the page has been manually refreshed is the shipping method checkbox selected again. 

The error only occurs after changing the payment method twice and remains after that. Only when changing the payment method for the first time after entering the checkout does everything still work normally.

**Comment by shopwareBot** [comment]
[public] Automated response: This issue is linked to the internal issue [https://shopware.atlassian.net/browse/NEXT-33085](https://shopware.atlassian.net/browse/NEXT-33085).

 

[created from NEXT-33085, comment 486120]

**Comment by shopwareBot** [comment]
[public] Hello,

 Within our company roadmap and work capacity, we try to address each bug or improvement request but admit that not each one will be resolved. To continue our culture of honesty and openness, we are closing this issue to focus on our roadmap on behalf of all Shopware users.

 If you feel like this issue is still impacting you, please create a new issue and let us know. Thank you again for your valuable feedback.



[created from NEXT-33085, comment 491888]
